pkgs/by-name/st/stract/package.nix

```
@@ -0,0 +1,39 @@
+{
+  pkgs,
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  curl,
+  fetchurl,
+  swaggerUi,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "stract";
+  version = "0-unstable-2024-09-14";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "StractOrg";
+    repo = "stract";
+    rev = "21d28ea86d9ff19ccb4c25b1bdde67e5ec302d79";
+    hash = "sha256-7Uvo5+saxwTMQjfDliyOYC6j6LbpMf/FiONfX38xepI=";
+  };
+
+  cargoHash = "sha256-7Skbeeev/xBAhlcyOsYpDJB9LnZpT66D0Fu1I/jIBso=";
+  cargoDepsName = pname;
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [
+    openssl
+    curl
+  ];
+
+  doCheck = false;
+  meta = {
+    description = "Stract is an open source web search engine hosted at stract.com targeted towards tinkerers and developers.";
```

```suggestion
    description = "Open source web search engine targeted towards tinkerers and developers";
```
Please read the [relevant manual chapter](https://nixos.org/manual/nixpkgs/stable/#chap-meta) for more information on the specifics of `meta` attributes.